pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java

```diff
@@ -232,6 +223,26 @@ public boolean onLongClick(View v) {
         mAnimate = mAnimatedType > -1;
     }
 
+    private int getMaxLength() {
+        for (InputFilter filter : getFilters()) {
+            if (filter instanceof InputFilter.LengthFilter) {
+                InputFilter.LengthFilter lengthFilter = ((InputFilter.LengthFilter) filter);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                    return lengthFilter.getMax();
+                } else {
+                    try {
+                        Field max = InputFilter.LengthFilter.class.getDeclaredField("mMax");
+                        max.setAccessible(true);
+                        return (int) max.get(lengthFilter);
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
Somehow in case exception we can not use expected length 🤔 What do you think about this approach? https://stackoverflow.com/a/60593495

Reply:
`maxlength` was initialized from parent class `TextView`. Why do we need to re-getting that out from property instead of getting them from filter? This exception is just for safety and so hard to happen I think